coordinate.go

```diff
@@ -418,3 +420,36 @@ func pingNode(addr string, method string) (time.Duration, error) {
 		return 0, pingErr
 	}
 }
+
+// Needed for cases of unregister->reregister, so the newly re-registered
+// node doesn't use the old status (which would stay for timed duration).
+// Also clear an inflightPings flag if present.
+func (a *Agent) checkNodeTracking(inCh nodeChannel) nodeChannel {
+	outCh := make(chan []*api.Node)
+	oldNodes := []*api.Node{}
```

Comment:
Does `oldNodes` need to be referenced outside of the goroutine? If not, wouldn't it be better to place it inside of the func scope to ensure that multithreading issues are less likely for future authors?

Reply:
Didn't catch that when I migrated it into a goroutine.
:+1: